app/src/main/java/tech/ula/utils/AndroidUtility.kt

```diff
@@ -398,4 +399,54 @@ class AcraWrapper {
     fun putCustomString(key: String, value: String) {
         ACRA.getErrorReporter().putCustomData(key, value)
     }
+}
+
+class UsageUtility(private val prefs: SharedPreferences) {
+    private val numberOfTimesOpenedKey = "numberOfTimesOpened"
+    private val userGaveFeedbackKey = "userGaveFeedback"
+    private val dateTimeFirstOpenKey = "dateTimeFirstOpen"
+    private val millisecondsInThreeDays = 259200000L
+    private val minimumNumberOfOpensBeforeReviewRequest = 15
+
+    fun addToNumberOfTimesOpened(): Int {
+        val numberTimesOpened = prefs.getInt(numberOfTimesOpenedKey, 1)
+
+        if (numberTimesOpened == 1) {
```

Comment:
I think you could move the the `if` statement into `with` to look a bit cleaner.

```
with (prefs.edit()) {
    putInt(numberOfTimesOpenedKey, numberTimesOpened + 1)
    if (numberTimesOpened == 1) putLong(firstTimeOpenedMillis, System.currentTimeMillis)
    apply()
}
```

Reply:
Oh yeah that looks a lot cleaner, I've changed it to move the if statement into the `with` block